src/hooks/useAssetLookup.ts

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable no-underscore-dangle */
+import { NETWORKS } from "config/constants";
+import {
+  PricedBalance,
+  SearchAssetResponse,
+  FormattedSearchAssetRecord,
+} from "config/types";
+import { formatAssetIdentifier, getAssetType } from "helpers/balances";
+import { isContractId } from "helpers/soroban";
+import useDebounce from "hooks/useDebounce";
+import { useState } from "react";
+import { handleContractLookup } from "services/backend";
+import { searchAsset } from "services/stellarExpert";
+
+export enum AssetLookupStatus {
+  IDLE = "idle",
+  LOADING = "loading",
+  SUCCESS = "success",
+  ERROR = "error",
+}
+
+interface UseAssetLookupProps {
+  network: NETWORKS;
+  publicKey?: string;
+  balanceItems: (PricedBalance & {
+    id: string;
+  })[];
+}
+
+export const useAssetLookup = ({
+  network,
+  publicKey,
+  balanceItems,
+}: UseAssetLookupProps) => {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [searchResults, setSearchResults] = useState<
+    FormattedSearchAssetRecord[]
+  >([]);
+  const [status, setStatus] = useState<AssetLookupStatus>(
+    AssetLookupStatus.IDLE,
+  );
+
+  const checkIfUserHasTrustline = (
```

Comment:
I'd suggest we rename it to something like `checkHasTrustline` or `checkTrustline` since an `User` could have multiple accounts/balances and this function only checks for a single account/balances. Does that make sense?